### ISSUE TITLE
Add Ninja configuration to Linux and Windows CI

### ISSUE THREAD

**Issue by andschwa** [Microsoft/openenclave#1283]
We should add a configuration to CI for building with the Ninja generator on Linux and Windows.

This will test not only that Ninja works, but also that parallel builds still work (since Ninja always runs in parallel).

Add an appropriate number of stages in Jenkins CI for each platform to use the Ninja build engine:
- [ ] Linux
- [x] Windows (once #1272 is merged) (Completed with Anita's PR #1415 )

**Comment by anitagov** [comment]
Changed the title to match the description that states that this is for Windows and Linux CI.
I am working on adding ninja to Windows CI.

**Comment by anitagov** [comment]
Changed the title to match the description that states that this is for Windows and Linux CI.
I am not working on the ninja Windows script but fixed the Windows CI PS script (test-config-build.ps1) to work correctly for Release mode with the 2 generators (part of another PR #1372  which is still not in master).

**Comment by anitagov** [comment]
Great if the Windows ninja script is prioritized before then Linux ninja work.

**Comment by ionutbalutoiu** [comment]
Sounds good @anitagov. Thanks for the updates!

**Comment by anitagov** [comment]
Issued a PR #1415 after spending 1/2 a day getting a new ACC Windows VM (plain vanilla) deployed, installed all the necessary tools manually and then proving that ninja worked with my changes to the Windows PS script.

@dragoshsch @ionutbalutoiu @johnkord - Can one of you install Visual Studio 2017 15.9.4 or later on the Windows CI systems so that we can get Ninja verified and this PR completed? The CMake tools will need to be installed (make sure that this individual component is selected before you install).

**Comment by ionutbalutoiu** [comment]
> Issued a PR #1415 after spending 1/2 a day getting a new ACC Windows VM (plain vanilla) deployed, installed all the necessary tools manually and then proving that ninja worked with my changes to the Windows PS script.
> 
> @dragoshsch @ionutbalutoiu @johnkord - Can one of you install Visual Studio 2017 15.9.4 or later on the Windows CI systems so that we can get Ninja verified and this PR completed? The CMake tools will need to be installed (make sure that this individual component is selected before you install).

I'll do that right now.

**Comment by anitagov** [comment]
Thanks @ionutbalutoiu . Seems like it built fine on one system/configuration and had 2 test issues but failed prematurely on the other (Debug Cross-compile config/machine).
A couple of questions - 1) Did you install Visual Studio 2017 or only the Build Tools?
2) On the debug Cross-compile build failure, the following error is reported.
Can you follow the instruction in the document https://github.com/Microsoft/openenclave/blob/master/docs/GettingStartedDocs/GettingStarted.Windows.md and verify that all the correct components are there. This points to issues with clang. Open a command window and do a where clang. It should report C:\Program Files\LLVM\bin\clang.exe. Please read the online guide to verify other components. Thanks.

-- The C compiler identification is Clang 7.0.1
CMake Error at C:/Program Files/CMake/share/cmake-.13/Modules/CMakeDetermineCompilerId.cmake:802 (message):
  The Clang compiler tool
    "C:/Program Files/LLVM/bin/clang-7.exe"
  targets the MSVC ABI but has a GNU-like command-line interface.  This is
  not supported.  Use 'clang-cl' instead, e.g.  by setting 'CC=clang-cl' in
  the environment.
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.13/Modules/CMakeDetermineCCompiler.cmake:113 (CMAKE_DIAGNOSE_UNSUPPORTED_CLANG)
  CMakeLists.txt:28 (project)

**Comment by ionutbalutoiu** [comment]
@anitagov I just replied in the PR with some minor discoveries that I got.

To directly answer your questions:

1) I installed only the [Visual C++ Tools for CMake](https://docs.microsoft.com/en-us/cpp/ide/cmake-tools-for-visual-cpp?view=vs-2017#installation) for the already installed `Visual Studio Build Tools 2017 15.9.6`. This comes with everything that we need (Ninja and CMake bundled).

    I thought about giving this a shot, instead of cluttering the build server with another Visual Studio installation. I might just get the full Visual Studio if you think this is really necessary.

2) Yep, all the requirements documented [here](https://github.com/Microsoft/openenclave/blob/master/docs/GettingStartedDocs/GettingStarted.Windows.md#prerequisites) seem to be fine. The PATH seems to be ok as well:
    ```
    PS C:\Users\oeadmin> get-command clang
    CommandType     Name                                               Version    Source
    -----------     ----                                               -------    ------
    Application     clang.exe                                          0.0.0.0    C:\Program Files\LLVM\bin\clang.exe

    PS C:\Users\oeadmin> get-command llvm-ar
    CommandType     Name                                               Version    Source
    -----------     ----                                               -------    ------
    Application     llvm-ar.exe                                        0.0.0.0    C:\Program Files\LLVM\bin\llvm-ar.exe


    PS C:\Users\oeadmin> get-command ld.lld
    CommandType     Name                                               Version    Source
    -----------     ----                                               -------    ------
    Application     ld.lld.exe                                         0.0.0.0    C:\Program Files\LLVM\bin\ld.lld.exe

    PS C:\Users\oeadmin> clang.exe --version
    clang version 7.0.1 (tags/RELEASE_701/final)
    Target: x86_64-pc-windows-msvc
    Thread model: posix
    InstalledDir: C:\Program Files\LLVM\bin

    PS C:\Users\oeadmin> get-command bash
    CommandType     Name                                               Version    Source
    -----------     ----                                               -------    ------
    Application     bash.exe                                           2.19.1.1   C:\Program Files\git\bin\bash.exe

    PS C:\Users\oeadmin> get-command ocaml
    CommandType     Name                                               Version    Source
    -----------     ----                                               -------    ------
    Application     ocaml.exe                                          0.0.0.0    C:\Program Files\ocpwin64\4.02.1+ocp1-msvc64-20160113\bin\ocaml.exe
    ```

**Comment by anitagov** [comment]
Thanks @ionutbalutoiu - Let's continue to discuss this in my PR. I just updated it with a few comments.
